scriptmodules/libretrocores/lr-mame2003.sh

```diff
@@ -47,27 +47,29 @@ function configure_lr-mame2003() {
 
     local mame_dir
     local mame_sub_dir
+
+    # create subfolders to prepopulate - mame2003 now generates other folders when needed
     for mame_dir in arcade mame-libretro; do
         mkRomDir "$mame_dir"
         mkRomDir "$mame_dir/$name"
         ensureSystemretroconfig "$mame_dir"
 
-        for mame_sub_dir in cfg ctrlr diff hi inp memcard nvram snap; do
-            mkRomDir "$mame_dir/$name/$mame_sub_dir"
+        # currently nvram is the only folder to repopulate
+        for mame_sub_dir in nvram; do
+            mkRomDir "$romdir/$mame_dir/$name/$mame_sub_dir"
```

Comment:
Why have you added $romdir ? Why are the other folders removed ? Looping through a single folder doesn't make sense.

Reply:
mame_dir is mame-libretro
 romdir is /home/pi/RetroPie/roms

i was just using the full path the other dirs where  removed because mame makes them itself

Reply:
MkRomDir adds $romdir already so I don't think the above will work correctly.

Reply:
@joolswills ah i see what you mean i should have used mkdir in that statement i see what was wrong with that. Thanks for the clarification.

Reply:
all that remains to be added is 

cp -r "$md_inst/metadata/nvram" "$romdir/$mame_dir/$name/$nvram"

unless you have a tidier way of doing it

Reply:
just for some clarification 

/libretro savefile dir/mame2003-plus/diff/
/libretro savefile dir/mame2003-plus/nvram/
/libretro savefile dir/mame2003-plus/hi/
/libretro savefile dir/mame2003-plus/cfg/
/libretro savefile dir/mame2003-plus/memcard/

Im not sure where retropie sets the libretro savefile dir the contents should be going into this dir im assuing its been set to the content dir